src/process.rs

```diff
@@ -179,14 +179,10 @@ pub(crate) fn spawn_program_on_event(
             env.insert("PLAYER_EVENT", "shuffle_changed".to_string());
             env.insert("SHUFFLE", shuffle.to_string());
         }
-        PlayerEvent::RepeatChanged { repeat } => {
+        PlayerEvent::RepeatChanged { context, track } => {
             env.insert("PLAYER_EVENT", "repeat_changed".to_string());
-            let val = match repeat {
-                true => "all",
-                false => "none",
-            }
-            .to_string();
-            env.insert("REPEAT", val);
+            env.insert("REPEAT", format!("{:?}", context));
+            env.insert("REPEAT_TRACK", format!("{:?}", track));
```

Comment:
To be in line with the previous behaviour, maybe something like
```suggestion
            let val = match (context, track) {
                (_, true) => "track",
                (true, false) => "all",
                (false, false) => "none",
            }
            .to_string();
            env.insert("REPEAT", val);
```
could work? (not tested)